devbox.go

```diff
@@ -30,7 +30,8 @@ const (
 	configFilename = "devbox.json"
 
 	// profileDir contains the contents of the profile generated via `nix-env --profile profileDir <command>`
-	profileDir = ".devbox/profile"
+	// Instead of using directory, prefer using the devbox.profileDir() function that ensures the directory exists.
+	profileDir = ".devbox/nix-profile/profile"
```

Comment:
move it under `/gen`?

Reply:
I initially felt we should reserve `gen` for generated files....maybe we should discuss the internal structure of `.devbox` folder.